yb-voyager/src/queryissue/queryissue.go

```diff
@@ -120,6 +126,31 @@ func (p *ParserIssueDetector) getAllIssues(query string) ([]issue.IssueInstance,
 	return issues, nil
 }
 
+func (p *ParserIssueDetector) GetPercentTypeSyntaxIssues(query string) ([]issue.IssueInstance, error) {
+	parseTree, err := queryparser.Parse(query)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing the query-%s: %v", query, err)
+	}
+
+	objType, objName := queryparser.GetObjectTypeAndObjectName(parseTree)
+	typeNames, err := queryparser.GetAllTypeNamesInPlpgSQLStmt(query)
+	if err != nil {
+		return nil, fmt.Errorf("error getting type names in PLPGSQL: %v", err)
+	}
+
+	if queryparser.IsFunctionObject(parseTree) {
```

Comment:
add the caveats comment here as well, so we know why we're making additional function calls after calling `GetAllTypeNamesInPlpgSQLStmt`